src/tristan/command_line/images.py

```diff
@@ -534,6 +535,116 @@ def multiple_sequences_cli(args):
     print(f"Images written to\n\t{output_nexus_pattern or out_file_pattern}")
 
 
+def gated_images_cli(args):
+    """Utility to bin events into a sequence of images according to a gating signal."""
+    write_mode = "w" if args.force else "x"
+    output_file = check_output_file(args.output_file, args.stem, "images", args.force)
+
+    input_nexus = args.data_dir / f"{args.stem}.nxs"
+    if not input_nexus.exists():
+        print(
+            "Could not find a NeXus file containing experiment metadata.\n"
+            "Resorting to writing raw image data without accompanying metadata."
+        )
+
+    image_size = args.image_size or determine_image_size(input_nexus)
+
+    raw_files, _ = data_files(args.data_dir, args.stem)
+
+    # If gate_close isn't specified, default to the complementary signal to gate_open.
+    gate_open = triggers.get(args.gate_open)
+    gate_close = triggers.get(args.gate_close) or gate_open ^ (1 << 5)
+
+    with latrd_data(raw_files, keys=cue_keys) as cues_data:
+        print("Finding detector shutter open and close times.")
+        with ProgressBar():
+            start, end = find_start_end(cues_data)
+
+        print("Finding gate signal times.")
+        open_times = cue_times(cues_data, gate_open, after=start)
```

Comment:
For `close_times`, I guess the check should in theory be before the `end` timestamp, corresponding to shutter close, instead of after `start`.
In any case, we still have to find a way to deal with the corner case of the first/last cue timestamp being a handful of ns befor/after the shutter.